hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/EntitySelectFetchInitializerBuilder.java

```diff
@@ -147,9 +147,8 @@ else if ( creationState.isDynamicInstantiation() ) {
 	}
 
 	private static boolean canBatchInitializeBeUsed(EntityPersister entityPersister) {
-		if ( entityPersister.getRepresentationStrategy().getProxyFactory() == null
-				&& entityPersister.hasSubclasses() ) {
-			// We cannot neither create a proxy nor instantiate the entity because we don't know the concrete type
+		if ( entityPersister.getRepresentationStrategy().getProxyFactory() == null ) {
+			// We cannot create a proxy
 			return false;
 		}
 		return true;
```

Comment:
Maybe the original condition should have used an `||` instead of `&&`? How about this simplification?

```suggestion
		return entityPersister.getRepresentationStrategy().getProxyFactory() != null && !entityPersister.hasSubclasses();
```

Or am I missing something?

Reply:
it can be 
`return entityPersister.getRepresentationStrategy().getProxyFactory() != null;`

Reply:
we need to create a proxy in order to use the `BATCH_INITIALIZE`